src/components/apply/avatar-radio.tsx

```diff
@@ -0,0 +1,58 @@
+import { ChangeEvent, useState } from "react";
+import Avatar from "./avatar";
+
+type AvatarType =
+  | "Wildlife Wanderer"
+  | "City Cruiser"
+  | "Foodie Fanatic"
+  | "Beach Bum"
+  | null
+  | undefined;
+
+interface AvatarRadioProps {
+  value: AvatarType;
+  onChange: (avatar: AvatarType) => void;
+}
+
+const avatarOptions: AvatarType[] = [
+  "Wildlife Wanderer",
+  "City Cruiser",
+  "Foodie Fanatic",
+  "Beach Bum",
+];
+
+const AvatarRadioGroup = ({ value, onChange }: AvatarRadioProps) => {
+  const [selectedAvatar, setSelectedAvatar] = useState<AvatarType | null>(
+    value,
+  );
+
+  const onSelect = (e: ChangeEvent<HTMLInputElement>) => {
+    const avatar = e.target.value as AvatarType;
+    setSelectedAvatar(avatar);
+    onChange(avatar);
+  };
+
+  return (
+    <div className="grid grid-cols-2 gap-8">
+      {avatarOptions.map((option, index) => (
+        <label key={index} className="flex w-full cursor-pointer">
+          <input
+            type="radio"
+            name="avatar"
+            checked={selectedAvatar === option}
+            onChange={onSelect}
+            className="hidden"
+          />
+          <Avatar
+            avatar={option}
+            className={`transition-opacity duration-200 ${
+              selectedAvatar === option ? "opacity-100" : "opacity-50"
+            }`}
+          />
+        </label>
+      ))}
+    </div>
```

Comment:
I like the structure of the component, but it doesn't work :// Take a look at RadioButtonItem in components/ui/radio-group.tsx (should be very similar except just change the styling of the item to include the images, and to reflect the button style)

Reply:
changed